projects/dynamicforms/src/core/services/data/form-data.service.ts

```diff
@@ -32,17 +34,16 @@ export class FormDataService {
   }
 
   saveFormData(formData: YFormData) {
-    this.userIdService
-      .getUserId()
-      .pipe(take(1))
-      .subscribe(occUserId => {
-        this.store.dispatch(
-          new fromAction.SaveFormData({
-            formData: formData,
-            userId: occUserId,
-          })
-        );
-      })
+    this.oboCustomerService
```

Comment:
Maybe it would be a good time to refactor this so it's not subscribing and unsubscribing here in the service. This should just return an observable and be subscribed to in the component that is consuming it. I see it's only being used in form.component, so it shouldn't be a huge refactor.